### PR TITLE
Add Header

### PR DESCRIPTION
**Pull request by scouillard**
## Description

Add first iteration of Header.
Replace the CurrentUser component to see if User is signed in or not.

If User is signed in, User will see his name on top right corner, with dropdown menu.

If User is signed out, User will see a Sign In and a Sign Up button.

## Testing Steps

Pull test and make sure everything works as before.
Try in incognito when User is signed off.

## Screenshots (if appropriate):

1. When User is signed in:

image

2. When User is signed out:

image

### CODE REVIEW

File: app/javascript/components/shared/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          />
+        </Navbar.Brand>
+
+        {currentUser?.signed_in
+          ? (
+            <NavDropdown title={currentUser?.name} id="basic-nav-dropdown">
+              <NavDropdown.Item as={Link} to="/">Profile</NavDropdown.Item>
+              <NavDropdown.Item as={Link} to="/">Need help?</NavDropdown.Item>
+              <NavDropdown.Divider />
+              <NavDropdown.Item as={Link} to="/" onClick={() => console.log("signOut")}>Sign Out</NavDropdown.Item>
```
Comment (hadicheaito1) [inline app/javascript/components/shared/Header.jsx:30]: Maybe change this to the actual sign_out request once the sign out PR is done

**Comment by hadicheaito1** [comment]
NOTE: Should wait for #3345 is merged

File: app/javascript/components/shared/Header.jsx (hunk 1 above)
Comment (scouillard) [reply app/javascript/components/shared/Header.jsx:30]: Yep, will do

**Review by KH-Amir-TN — APPROVED** [review]

File: app/javascript/components/shared/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { Navbar, NavDropdown, Stack } from 'react-bootstrap';
+import { Link } from 'react-router-dom';
+import { useAuth } from '../sessions/AuthContext';
+
+export default function Header() {
+  const currentUser = useAuth();
+
+  return (
+    <Navbar bg="light">
+      <Container>
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/javascript/components/shared/Header.jsx:13]: Ah wait, I see what you did

File: app/javascript/components/shared/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              <NavDropdown.Item as={Link} to="/">Profile</NavDropdown.Item>
+              <NavDropdown.Item as={Link} to="/">Need help?</NavDropdown.Item>
+              <NavDropdown.Divider />
+              <NavDropdown.Item as={Link} to="/" onClick={() => console.log("signOut")}>Sign Out</NavDropdown.Item>
+            </NavDropdown>
+          )
+
+          : (
+            <Stack direction="horizontal">
+              <Link to="/sign_in">
```
[the remark this answers is not in the document]
Comment (scouillard) [reply app/javascript/components/shared/Header.jsx:36]: Will do.
Waiting on #3345 to push changes.

File: app/javascript/components/shared/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { useAuth } from '../sessions/AuthContext';
+
+export default function Header() {
+  const currentUser = useAuth();
+
+  return (
+    <Navbar bg="light">
+      <Container>
+        <Navbar.Brand as={Link} to="/">
+          <img
```
Comment (KH-Amir-TN) [inline app/javascript/components/shared/Header.jsx:15]: I think using the react bootstrap `Image` component is more adequate.

File: app/javascript/App.jsx [hunk 5]
```diff
@@ -1,13 +1,15 @@
 import React from 'react';
 import { Container } from 'react-bootstrap';
 import { Outlet } from 'react-router-dom';
-import CurrentUser from './components/user/CurrentUser';
+import Header from './components/shared/Header';
```
Comment (KH-Amir-TN) [inline app/javascript/App.jsx:4]: Good point, I think changing the `/components/stylings -> /components/shared` may be more adequate.
What do you think guys?

File: app/javascript/components/shared/Header.jsx (hunk 4 above)
Comment (scouillard) [reply app/javascript/components/shared/Header.jsx:17]: I got this from React Bootstrap demo.

https://react-bootstrap.github.io/components/navbar/

**Review by farhatahmad — APPROVED** [review]

File: app/javascript/components/shared/Header.jsx (hunk 4 above)
Comment (KH-Amir-TN) [reply app/javascript/components/shared/Header.jsx:17]: Yes, but there exists this [component ](https://react-bootstrap.netlify.app/components/images/#images) it gives some extra stuff like shape that may be usueful.

**Pull request merged**